packages/amplify-graphql-relational-transformer/src/utils.ts

```diff
@@ -173,14 +195,37 @@ function getIndexName(directive: DirectiveNode): string | undefined {
   }
 }
 
-export function getConnectionAttributeName(type: string, field: string) {
-  return toCamelCase([type, field, 'id']);
+/**
+ *
+ */
+export function getConnectionAttributeName(featureFlags: FeatureFlagProvider, type: string, field: string, relatedTypeField: string) {
+  const nameSuffix = featureFlags.getBoolean('useFieldNameForPrimaryKeyConnectionField') ? relatedTypeField : 'id';
+  return toCamelCase([type, field, nameSuffix]);
+}
+
+/**
+ *
+ */
+export function getSortKeyConnectionAttributeName(featureFlags: FeatureFlagProvider, type: string, field: string, relatedTypeField: string) {
```

Comment:
`featureFlags` is not used in this function

Reply:
thanks, forgot to remove after some refactor, updated